### PR TITLE
Discoverable repos

### PR DESCRIPTION
**Pull request by alfredodeza**

### CODE REVIEW

File: shaman/tests/controllers/test_repos.py [hunk 1]
```diff
@@ -48,8 +63,8 @@ def setup(self):
     def test_get_not_allowed(self, session):
```
Comment (andrewschoen) [inline shaman/tests/controllers/test_repos.py]: You'll need to update the name of this test now as it does not apply

File: shaman/controllers/repos/projects.py [hunk 2]
```diff
@@ -16,6 +16,12 @@ def __init__(self, project_name):
     def index(self):
         abort(405)
 
+    @index.when(method='GET', template='json')
+    def index_get(self):
+        if not self.project:
+            abort(404)
+        return self.project
```
Comment (andrewschoen) [inline shaman/controllers/repos/projects.py]: Shouldn't this endpoint return information about what types of repos the project has?

Comment (alfredodeza) [inline shaman/controllers/repos/projects.py]: Goo question. If we do it that way we wouldn't have an endpoint where we could consume the actual repo information though. Unless I add a field or separate both. Something like:

    dict(
        "project": self.project,
        "repos": self.project.repos
     )

Or maybe try overload the json coming from the Project model?

File: shaman/controllers/repos/flavors.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    def __init__(self, flavor_name):
+        self.flavor_name = flavor_name
+        self.project = Project.query.get(request.context['project_id'])
+        self.repo = Repo.query.filter_by(
+            project=self.project,
+            ref=request.context['ref'],
+            sha1=request.context['sha1'],
+            distro=request.context['distro'],
+            distro_version=request.context['distro_version'],
+            flavor=flavor_name).first()
```
Comment (andrewschoen) [inline shaman/controllers/repos/flavors.py]: I'm not sure you can do ``.first()`` here. There is a small chance we might get the exact same repo built on two different chacra nodes.

**Pull request merged**